### PR TITLE
Improve NoteForm buttons and blur background component

### PR DESCRIPTION
**Pull request by dForDeveloper**
I added a discard button on the NoteForm at the `/new-note` path. The button links to `/`. I also made the NoteContainer blur out and not register click events when the NoteForm is rendered. Lastly, I added more tests to achieve 100% test coverage of the entire app. 💯
screen shot 2019-02-17 at 1 19 07 pm

### CODE REVIEW

File: src/containers/App/App.js [hunk 1]
```diff
@@ -13,33 +13,33 @@ export class App extends Component {
...
     ) : <Error404 />;
   }
 
+  getNewNoteRoute = ({ match }) => {
+    return [
+      <NoteContainer isDisabled={true} key="NoteContainer"/>,
+      <NoteForm match={match} key="NoteForm"/>
+    ]
+  }
+
```
Comment (whitneyfirth) [inline src/containers/App/App.js:33]: Great idea to break this out into it's own function!! The render looks nice and clean, love it.

File: src/containers/App/App.js [hunk 2]
```diff
@@ -13,33 +13,33 @@ export class App extends Component {
...
     const { id } = match.params;
-    const currentNote = this.props.notes.find( note => note.id === id );
+    const currentNote = notes.find( note => note.id === id );
     return currentNote ? (
-      <Fragment>
-        <NoteContainer/>
-        <NoteForm {...currentNote} match={match}/>
-      </Fragment>
+      [
+        <NoteContainer isDisabled={true} key="NoteContainer" />,
```
Comment (whitneyfirth) [inline src/containers/App/App.js:21]: I'm glad we have this disabled now - makes for such a better user experience. Nice work.

File: src/containers/App/App.test.js [hunk 3]
```diff
@@ -42,11 +42,22 @@ describe('App', () => {
...
+      const mockMatch = { params: { id: 'hghgfjhg' }, path: 'notes/hghgfjhg' };
       const result = wrapper.instance().getNotesRoute({ match: mockMatch });
       const wrapper404 = shallow(result);
       expect(wrapper404.find('.Error404')).toHaveLength(1);
     });
+
+    it('should return an array of JSX', () => {
+      const result = wrapper.instance().getNewNoteRoute({ match: mockMatch });
+      expect(result).toHaveLength(2);
+    });
```
Comment (whitneyfirth) [inline src/containers/App/App.test.js:60]: Good tests for these two new functions!  👍

Comment (trbachmann) [reply src/containers/App/App.test.js:60]: Agreed! Glad you were able to figure out the tests for the other part of these functions – great job!

File: src/containers/NoteContainer/NoteContainer.scss [hunk 4]
```diff
@@ -1,3 +1,8 @@
+.NoteContainer--disabled {
```
Comment (trbachmann) [inline src/containers/NoteContainer/NoteContainer.scss:1]: Really like how this looks when the user is on the note form, nice work! 👍

File: src/containers/NoteContainer/NoteContainer.test.js [hunk 5]
```diff
@@ -4,8 +4,24 @@ import { NoteContainer, mapStateToProps } from './NoteContainer';
...
 describe('NoteContainer', () => {
+  const mockProps = {
+    notes: mockNotes,
+    isLoading: false,
+    isDisabled: false
+  }
+
   it('should match the snapshot', () => {
-    let wrapper = shallow(<NoteContainer notes={mockNotes} isLoading={false}/>)
+    let wrapper = shallow(<NoteContainer {...mockProps} />);
```
Comment (trbachmann) [inline src/containers/NoteContainer/NoteContainer.test.js:14]: Good tests to ensure the NoteContainer is disabled

File: src/containers/App/App.js (hunk 2 above)
Comment (trbachmann) [reply src/containers/App/App.js:21]: Agreed! Great job 🙂

File: src/containers/NoteContainer/NoteContainer.js [hunk 6]
```diff
@@ -20,19 +20,19 @@ export const NoteContainer = ({ notes, isLoading }) => {
     550: 1
   };
 
+  const disabledClass = isDisabled ? '--disabled' : '';
+
```
Comment (whitneyfirth) [inline src/containers/NoteContainer/NoteContainer.js:24]: Nice conditional rendering for our blurred out / disabled background.

**Review by trbachmann — APPROVED** [review]
Great work Jeo! Really glad the NoteContainer is disabled now when the user is on the NoteForm. Awesome work getting the app to 100% test coverage too!! 🔥💯

File: src/containers/NoteContainer/NoteContainer.scss [hunk 7]
```diff
@@ -1,3 +1,8 @@
...
+  pointer-events: none;
```
Comment (whitneyfirth) [inline src/containers/NoteContainer/NoteContainer.scss:2]: This is great, I haven't used this before!

File: src/containers/NoteForm/NoteForm.js [hunk 8]
```diff
@@ -138,20 +138,28 @@ export class NoteForm extends Component {
...
     const { path } = this.props.match;
     return (
       <div className='NoteForm'>
         {this.getTitleInput()}
         {this.getListItems()}
         {this.getNewListItemInput()}
-        <button className='NoteForm--submit' onClick={this.handleSubmit}>
+        <button
+          className='NoteForm--submit'
+          disabled={title.trim() === '' || listItems.length === 0}
```
Comment (whitneyfirth) [inline src/containers/NoteForm/NoteForm.js:151]: Nice check for disabling the button! 🔥

File: src/containers/NoteForm/NoteForm.test.js [hunk 9]
```diff
@@ -121,6 +127,12 @@ describe('NoteForm', () => {
       expect(wrapperInput.find(inputClass)).toHaveLength(1);
       expect(wrapperInput.find('[value="Workout"]')).toHaveLength(1);
     });
+
+    it('should set state if the title input changes', () => {
+      const mockEvent = { target: { value: 'A' } };
+      wrapperNewNote.find('.NoteForm--title').simulate('change', mockEvent);
+      expect(wrapperNewNote.state('title')).toEqual('A');
+    });
```
Comment (whitneyfirth) [inline src/containers/NoteForm/NoteForm.test.js:135]: Great tests Jeo!

**Review by whitneyfirth — APPROVED** [review]
Awesome work Jeo 🙌

**Pull request merged**